### PR TITLE
Housekeeping

### PR DESCRIPTION
**Pull request by jjerphan**
**Main changes**:
 -  Fix typo: agregation to aggregation
 - Build and install pdist_aggregation from source
 - Use python 3.9 for benchmarks
 - Add a minimal README 
 
@mbatoul: feel free to add anything (CI setup, etc.)

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -15,3 +16,5 @@ dependencies:
       - --pre
       - --index-url https://pypi.anaconda.org/scipy-wheels-nightly/simple
       - scikit-learn
+      # Build and install pdist_aggregation from source
+      - -e src -v
```
Comment (mbatoul) [inline environment.yml]: The `-v` command line option does not seem to work in this context:

```
Installing pip dependencies: | Ran pip subprocess with arguments:
['/Users/mathisbatoul/opt/miniconda3/envs/pdist_aggregation/bin/python', '-m', 'pip', 'install', '-U', '-r', '/Users/mathisbatoul/dev/inria/pdist_aggregation/condaenv.vfesp2ks.requirements.txt']
Pip subprocess output:

Pip subprocess error:
Usage: __main__.py [options]

ERROR: Invalid requirement: -e src -v
__main__.py: error: no such option: -v


failed

CondaEnvException: Pip failed
```

Am I missing something?

Comment (jjerphan) [reply environment.yml]: I got the same problem, which I fixed in https://github.com/scikit-learn-inria-fondation/pdist_aggregation/commit/7b03d7306833293200fb3421df0e8676993717a4 for #3.

You can fix it here by cheery-picking this commit. 🙂

Comment (mbatoul) [reply environment.yml]: Done. Thanks!

File: src/setup.py [hunk 2]
```diff
@@ -5,8 +5,8 @@
 
 extensions = [
     Extension(
-        "pdist_agregation",
-        sources=["_pdist_agregation.pyx"],
+        "pdist_aggregation",
+        sources=["_pdist_aggregation.pyx"],
         include_dirs=[numpy.get_include()],
         extra_compile_args=['-fopenmp'],
         extra_link_args=['-fopenmp'],
```
Comment (mbatoul) [inline src/setup.py:12]: I got this error during the build: `clang: error: unsupported option '-fopenmp'` (on MacOS).

A hotfix for me was to add the `-Xpreprocessor` flag.

```
extensions = [
    Extension(
        "pdist_aggregation",
        sources=["_pdist_aggregation.pyx"],
        include_dirs=[numpy.get_include()],
        extra_compile_args=["-Xpreprocessor", "-fopenmp"],
        extra_link_args=["-Xpreprocessor", "-fopenmp"],
        define_macros=[("NPY_NO_DEPRECATED_API", "NPY_1_7_API_VERSION")],
    )
]
```

Comment (jjerphan) [reply src/setup.py:12]: Did OpenMP come when installing? You can check with:
```bash
conda list -n pdist_aggregation
```

Comment (mbatoul) [reply src/setup.py:12]: It seems that it did:

```
$ conda list -n pdist_aggregation | grep openmp
libopenblas               0.3.15          openmp_h5e1b9a4_0    conda-forge
llvm-openmp               11.1.0               hda6cdc1_1    conda-forge
```

Comment (jjerphan) [reply src/setup.py:12]: Which `clang` are you using?

See this comment regarding `llvm` and OpenMP on Mac: https://github.com/microsoft/LightGBM/issues/3#issuecomment-332025880

Comment (mbatoul) [reply src/setup.py:12]: I made it work using the [`compilers`](https://anaconda.org/conda-forge/compilers) package (see [64da803](https://github.com/scikit-learn-inria-fondation/pdist_aggregation/pull/6/commits/64da8033cf798148519aa671d51c386b7a9a41e1)).

Thanks @ogrisel!

Apparently the system `clang` in does not support `OpenMP` by default on MacOS.

**Comment by jjerphan** [comment]
I let you merge this PR if it's fine for you, @mbatoul.

**Pull request merged**